### PR TITLE
Remove responsibility for managing state from [ButtonGroup]

### PR DESCRIPTION
**Pull request by yujong-lee**
## Description

1. [`given`](https://github.com/tatyshev/given2) and [`context`](https://github.com/negativetwelve/jest-plugins/tree/master/packages/jest-plugin-context) are awesome libraries to write readable test codes. Although I use only `context` here, they usually work together.  So I add both to commit.
2. [ButtonGroup] doesn't need to be responsible for managing selected option. It makes testing hard(in general), and reduces reusability and flexibility of component.

## Related Issues
#52

## Tests
I update test code first. Then modify [ButtonGroup] to fulfill tests.
So changes in test code represents what I did in [ButtonGroup].

## Checklist
- [x] I read the [Contributor Guide](https://github.com/dooboolab/dooboo-ui/blob/master/CONTRIBUTING.md) and followed the process outlined there for submitting PRs.
- [x] Run `yarn test` or `yarn test -u` if you need to update snapshot.
- [x] Run `yarn lint`
- [x] I am willing to follow-up on review comments in a timely manner.

### CODE REVIEW

File: docs/src/components/ButtonGroup/ButtonGroup.stories.mdx [hunk 1]
```diff
@@ -29,7 +29,7 @@ import {ButtonGroup} from 'dooboo-ui';
 | data                   | `string[]`                 | List of strings to be rendered in each button. |
 | color?                 | `string`                   | Represents main color.                         |
 | onPress?               | `(i: number) => void`      | `callback` for press event.                    |
-| initialIndex?          | `number`                   | Initial `index` to be selected.                |
+| selected?              | `number`                   | selected `index`.                              |
```
Comment (hyochan) [inline docs/src/components/ButtonGroup/ButtonGroup.stories.mdx:32]: It feels like `selected` props should be boolean. How about renaming it to `selectedIndex`?

**Review by hyochan — CHANGES_REQUESTED** [review]

**Comment by hyochan** [comment]
@yujong-lee Yeah [jest-plugin-context](https://www.npmjs.com/package/jest-plugin-context) looks awesome. But I am afraid that this isn't been maintained for 3 years 🤔

**Comment by hyochan** [comment]
Also we need to add `@types/jest-plugin-context` for typescript

File: docs/src/components/ButtonGroup/ButtonGroup.stories.mdx (hunk 1 above)
Comment (yujong-lee) [reply docs/src/components/ButtonGroup/ButtonGroup.stories.mdx]: Yes. you are definitely right!

**Comment by yujong-lee** [comment]
Ok. I rename variable and install `@types/jest-plugin-context`.

There's few thing I want to share.

1. `context` is just aliases of `describe`.
See [source code](https://github.com/negativetwelve/jest-plugins/blob/master/packages/jest-plugin-context/src/context.js). So In this case, I think the fact that this library is not maintained for 3 years doesn't really matters.

2. But, `given2` is more complex library, and it's not maintaining these days too.
So I find two alternative library. [givens](https://github.com/enova/givens) and [bdd-lazy-var](https://github.com/stalniy/bdd-lazy-var). 

`givens` : similar interface with `given2`. Which is familiar to me and other RSpec-background-developers. Auther said it has advantages over `given2`and `bdd-lazy-var`.  But I haven't used/heard of it so it need some testing. It is also maintained by single author.

`bdd-lazy-var` : different interface with `given2`. Has more feature than `given2`, but they are not crucial.(in my opinion.) Good documentation. Many contributors. Updated(except bot) less than 1 year. It has built-in `context` aliases.

I'm not confident to choose one now. (`jest-plugin-context + given2 or givens` / `add-lazy-var`)
So what I suggest is add this talk to discussion. remove `given2` from this PR, and choose among these options when it is needed. (It will be needed soon).

File: main/__tests__/ButtonGroup.test.tsx [hunk 2]
```diff
@@ -10,11 +10,11 @@ describe('[ButtonGroup]', () => {
 
   const handlePress = jest.fn();
 
-  const renderButtonGroup = (): RenderAPI =>
+  const renderButtonGroup = (index?: number): RenderAPI =>
     render(
       <ThemeProvider initialThemeType={themeType}>
         <ButtonGroup
-          initialIndex={0}
+          selected={index}
```
Comment (hyochan) [inline main/__tests__/ButtonGroup.test.tsx:17]: Looks like the props here is not changed yet `selectedIndex`

**Comment by hyochan** [comment]
> Ok. I rename variable and install `@types/jest-plugin-context`.
> 
> There's few thing I want to share.
> 
> 1. `context` is just aliases of `describe`.
>    See [source code](https://github.com/negativetwelve/jest-plugins/blob/master/packages/jest-plugin-context/src/context.js).   So In this case, I think the fact that this library is not maintained for 3 years doesn't really matters.
> 2. But, `given2` is more complex library, and it's not maintaining these days too.
>    So I find two alternative library. [givens](https://github.com/enova/givens) and [bdd-lazy-var](https://github.com/stalniy/bdd-lazy-var).
> 
> `givens` : similar interface with `given2`. Which is familiar to me and other RSpec-background-developers. Auther said it has [advantages](https://stackoverflow.com/a/59926984/14074974) over `given2`and `bdd-lazy-var`. But I haven't used/heard of it so it needs some testing. It is also maintained by single author.
> 
> `bdd-lazy-var` : different interface with `given2`. Has more feature than `given2`, but they are not crucial.(in my opinion.) Good documentation. Many contributors. Updated(except bot) less than 1 year. It has built-in `context` aliases.
> 
> I'm not confident to choose one now. (`jest-plugin-context + given2 or givens` / `bdd-lazy-var`)
> So what I suggest is add this talk to discussion. remove `given2` from this PR, and choose among these options when it is needed. (It will be needed soon).

It looks like a great discussion to start! Could you kindly open up a discussion?

File: main/__tests__/ButtonGroup.test.tsx (hunk 2 above)
Comment (yujong-lee) [reply main/__tests__/ButtonGroup.test.tsx]: Sorry for that. I made commit to fix this.

Comment (yujong-lee) [reply main/__tests__/ButtonGroup.test.tsx]: This happens due to insufficient test that didn't caught my mistake.
I update tests to test initial value of selectedIndex. and fix mistake.

**Comment by yujong-lee** [comment]
No problem!

**Review by hyochan — APPROVED** [review]
LGTM!

**Pull request merged**